src/components/Search/SearchViewInit.tsx

```diff
@@ -1,21 +1,18 @@
 import React from 'react';
 
-import { connect } from 'react-redux';
+import { connect, useDispatch } from 'react-redux';
 
-import { store } from '../../common/store';
 import { fetchRecentRevisions } from '../../thunks/searchThunk';
 import { Repository, State } from '../../types/state';
 
 // component to fetch recent revisions when search view is loaded
-class SearchViewInit extends React.Component<SearchViewInitProps> {
-  componentDidMount() {
-    const { repository } = this.props;
-    store.dispatch(fetchRecentRevisions(repository));
-  }
-
-  render() {
-    return null;
-  }
+function SearchViewInit(props: SearchViewInitProps) {
+  const dispatch = useDispatch();
+  const { repository } = props;
+  React.useEffect(() => {
```

Comment:
converting this to a functional component and using `useEffect` instead of `componentDidMount` seemed to resolve this. this should have the same effect, only it is called after render instead of before: https://reactjs.org/docs/hooks-reference.html#useeffect